testsuite/run_forge.sh

```diff
@@ -76,17 +76,34 @@ set_forge_namespace() {
 
 # Set an image tag to use
 set_image_tag() {
-    if [ -z "$IMAGE_TAG" ]; then
-        IMAGE_TAG_DEFAULT=$(git rev-parse HEAD)
-        echo "IMAGE_TAG not set, defaulting to current HEAD commit as tag: ${IMAGE_TAG_DEFAULT}"
-        IMAGE_TAG=${IMAGE_TAG_DEFAULT}
-    fi
     echo "Ensure image exists"
-    img=$(aws ecr describe-images --repository-name="aptos/validator" --image-ids=imageTag=$IMAGE_TAG)
-    if [ $? != 0 ]; then
-        echo "IMAGE_TAG does not exist in ECR: ${IMAGE_TAG}. Make sure your commit has been pushed to GitHub previously."
-        echo "If you're trying to run the code from your PR, apply the label 'CICD:build-images' and wait for the builds to finish."
-        exit 1
+    # if IMAGE_TAG not set, check the last few commits on HEAD
+    if [ -z "$IMAGE_TAG" ]; then
+        echo "IMAGE_TAG not set, trying the latest commits before HEAD"
+        commit_threshold=5
+        for i in $(seq 0 $commit_threshold); do
```

Comment:
Can you explain why we need to this? This might be confusing and error prone if the test is running on an image tag, which is not the latest?

Reply:
Images are currently built in two places:
* When PRs land, on the PR branch
* After a PR lands, and push to `main`

In the first case, the git sha1 may not match that of `main` after it lands. This is why we build another image after the PR lands, which is the second case.

In the second case, there's a chance that this GHA is triggered before the image finishes building, so the latest commit will not have an image associated with it yet. In this case, we do this logic of checking each previous commit and checking if there is a matching image.